Android/android.py

```diff
@@ -74,41 +78,62 @@ def subdir(*parts, create=False):
 
 def run(command, *, host=None, env=None, log=True, **kwargs):
     kwargs.setdefault("check", True)
+
     if env is None:
         env = os.environ.copy()
-    original_env = env.copy()
-
     if host:
-        env_script = ANDROID_DIR / "android-env.sh"
-        env_output = subprocess.run(
-            f"set -eu; "
-            f"HOST={host}; "
-            f"PREFIX={subdir(host)}/prefix; "
-            f". {env_script}; "
-            f"export",
-            check=True, shell=True, text=True, stdout=subprocess.PIPE
-        ).stdout
-
-        for line in env_output.splitlines():
-            # We don't require every line to match, as there may be some other
-            # output from installing the NDK.
-            if match := re.search(
-                "^(declare -x |export )?(\\w+)=['\"]?(.*?)['\"]?$", line
-            ):
-                key, value = match[2], match[3]
-                if env.get(key) != value:
-                    print(line)
-                    env[key] = value
-
-        if env == original_env:
-            raise ValueError(f"Found no variables in {env_script.name} output:\n"
-                             + env_output)
+        env.update(android_env(host))
 
     if log:
         print(">", " ".join(map(str, command)))
     return subprocess.run(command, env=env, **kwargs)
 
 
+def print_env(context):
+    android_env(getattr(context, "host", None))
+
+
+def android_env(host):
+    if host:
+        prefix = subdir(host) / "prefix"
+    else:
+        prefix = ANDROID_DIR / "prefix"
+        sysconfigdata_files = prefix.glob("lib/python*/_sysconfigdata__android_*.py")
+        host = re.fullmatch(
+            r"_sysconfigdata__android_(.+).py", next(sysconfigdata_files).name
+        )[1]
```

Comment:
```suggestion
        sysconfigdata_files = prefix.glob("lib/python*/_sysconfigdata__android_*.py")
        sysconfigdata_file = next(sysconfigdata_files).name
        host = re.fullmatch(r"_sysconfigdata__android_(.+).py", sysconfigdata_file)[1]
```

Reply:
Done, with a small change to the variable names to distinguish more clearly between Paths and strings.